language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/InvocationNodeContextProvider.java

```diff
@@ -89,9 +89,25 @@ public void sort(BallerinaCompletionContext context, T node, List<LSCompletionIt
         }
         for (LSCompletionItem completionItem : completionItems) {
             if (completionItem.getType() == LSCompletionItem.CompletionItemType.NAMED_ARG) {
-                String sortText = SortingUtil.genSortText(1) +
-                        SortingUtil.genSortText(SortingUtil.toRank(context, completionItem));
+                NamedArgCompletionItem argCompletionItem = (NamedArgCompletionItem) completionItem;
+                Either<ParameterSymbol, RecordFieldSymbol> symbol = argCompletionItem.getParameterSymbol();
+                String sortText;
+                if (symbol.isRight()) {
+                    RecordFieldSymbol right = symbol.getRight();
+                    if (right.isOptional()) {
+                        sortText = SortingUtil.genSortText(1) + SortingUtil.genSortText(3);
+                    } else if (right.hasDefaultValue()) {
+                        sortText = SortingUtil.genSortText(1) + SortingUtil.genSortText(2);
+                    } else {
+                        sortText = SortingUtil.genSortText(1) +
```

Comment:
Is there a chance that a required field is assigned sort text `AC` while a field with default value gets assigned `AB`?

Reply:
Let's use `genSortText(1)` as the send part of the sort text as discussed offline.

Reply:
Fixed in [ad8e63e](https://github.com/ballerina-platform/ballerina-lang/pull/39057/commits/ad8e63e42f73fb8c6a8f9ef94aac7d999a199e51)